src/main/java/life/qbic/api/rest/v2/samples/SamplesControllerV2.java

```diff
@@ -55,22 +56,16 @@ public HttpResponse<?> moveSampleToStatus(@PathVariable String sampleCode,
     SampleStatusDto requestedStatus = statusChangeRequest.status();
     if (SampleStatusDto.METADATA_REGISTERED.equals(requestedStatus)) {
       sampleService.registerMetadata(sampleCode, validSince);
-
     } else if (SampleStatusDto.SAMPLE_RECEIVED.equals(requestedStatus)) {
       sampleService.receiveSample(sampleCode, validSince);
-
     } else if (SampleStatusDto.SAMPLE_QC_FAIL.equals(requestedStatus)) {
       sampleService.failQualityControl(sampleCode, validSince);
-
     } else if (SampleStatusDto.SAMPLE_QC_PASS.equals(requestedStatus)) {
       sampleService.passQualityControl(sampleCode, validSince);
-
     } else if (SampleStatusDto.LIBRARY_PREP_FINISHED.equals(requestedStatus)) {
       sampleService.prepareLibrary(sampleCode, validSince);
-
     } else if (SampleStatusDto.DATA_AVAILABLE.equals(requestedStatus)) {
       sampleService.provideData(sampleCode, validSince);
-
```

Comment:
Could this be moved into a seperate method (DtoToSampleStatus) in the SampleStatusDTOMapper class? Grouping this would help to not forget adaptions to the sample status imo.

Reply:
I thought about this but I do not think so. We call different methods here and do not simply map the status.

Reply:
Ah that makes sense 🧠